modules/storage/storage-account/main.bicep

```diff
@@ -0,0 +1,68 @@
+@description('Deployment Location')
+param location string
+
+@description('Name of Storage Account. Must be unique within Azure.')
+param name string = 'st${uniqueString(resourceGroup().id, subscription().id)}'
+
+@allowed([ 'new', 'existing' ])
+@description('Specifies whether to create a new Storage Account or use an existing one.')
+param newOrExisting string = 'new'
+
+@description('Name of Resource Group where the Storage Account will be deployed.')
+param resourceGroupName string = resourceGroup().name
+
+@description('ID of the subnet where the Storage Account will be deployed, if virtual network access is enabled.')
+param subnetID string = ''
+
+@description('Toggle to enable or disable virtual network access for the Storage Account.')
+param enableVNET bool = false
+
+@description('Toggle to enable or disable zone redundancy for the Storage Account.')
+param isZoneRedundant bool = false
+
+@description('Storage Account Type. Use Zonal Redundant Storage when able.')
+param storageAccountType string = isZoneRedundant ? 'Standard_ZRS' : 'Standard_LRS'
+
+var networkAcls = enableVNET ? {
+  defaultAction: 'Deny'
+  virtualNetworkRules: [
+    {
+      action: 'Allow'
+      id: subnetID
+    }
+  ]
+} : {}
+
+resource newStorageAccount 'Microsoft.Storage/storageAccounts@2022-09-01' = if (newOrExisting == 'new') {
```

Comment:
I'm not sure I understand why I would use a module like this to get a reference to an existing storage account. I understand this pattern when there is a larger deployment with several resource types, but since this is just a single module, it does not seem necessary.

Reply:
I've designed the module so that it can be used without the users having to worry about the newOrExisting mechanics. Instead, By handling this within the module, we can reduce the boiler plate code required to use the module, regardless of if the resource is new or existing.

# With internal handling of newOrExisting
```bicep
module storageAccount 'br/public:storage/storage-acount:0.0.1' = {
  name: 'mystorageaccount'
  params: {
    location: location
    name: name
    resourceGroupName: resourceGroupName
    isZoneRedundant: isZoneRedundant
    newOrExisting: newOrExisting
  }
}
var storageAccountId = storageAccount.outputs.id
```

# Without internal handling of newOrExisting
```bicep
module newStorageAccount 'br/public:storage/storage-acount:0.0.1' = if (newOrExisting == 'new') {
  name: 'mystorageaccount'
  params: {
    location: location
    name: name
    resourceGroupName: resourceGroupName
    isZoneRedundant: isZoneRedundant
  }
}

resource existingStorageAccount 'Microsoft.Storage/storageAccounts@2022-09-01' existing = if (newOrExisting == 'existing') {
  scope: resourceGroup(resourceGroupName)
  name: name
}
var storageAccountId = newOrExisting == 'new' ? newStorageAccount.outputs.id : existingStorageAccount.properties.id
```